### PR TITLE
docs add nixos install instructions

### PR DESCRIPTION
**Pull request by Janik-Haag**
Hi I created a small docs draft adding install instructions for NixOS since I initially wrote the woodpecker NixOS module which is now in stable for ~6 months and multiple people my self included actively use it. cc @ambroisie since we co-maintain it and he did a lot of improvement work on the module.

### CODE REVIEW

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Kubernetes backend
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:1]: Typo?

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+:::info
+Note that this module is not maintained by the woodpecker-developers
+:::
+
+The NixOS install is in theory quite similar to the binary install and supports multiple backends.
+In practice you specify the settings decoratively in your NixOS config and don't have to do any manual steps.
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:8]:
```suggestion
In practice you specify the settings declaratively in your NixOS config and don't have to do any manual steps.
```

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+```nix
+{ config
+, ...
+}:
+let
+  domain = "woodpecker.example.org";
+in
+{
+  # This automatically sets up certificates via let's encrypt
+  security.acme.certs."${domain}" = { };
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:21]: You also need to accept the terms of use IIRC.

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    enableACME = true;
+    forceSSL = true;
+    locations."/" = {
+      proxyPass = "http://localhost:3007";
+      proxyWebsockets = true;
+    };
+  };
+
+  # This is needed so the agents can reach the server via grpc
+  networking.firewall.allowedTCPPorts = [ 9000 ];
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:34]: Not necessary if the agents are on the same host as the server.

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  # This automatically sets up certificates via let's encrypt
+  security.acme.certs."${domain}" = { };
+
+  # Setting up a nginx proxy that handles tls for us
+  services.nginx.virtualHosts."${domain}" = {
+    enableACME = true;
+    forceSSL = true;
+    locations."/" = {
+      proxyPass = "http://localhost:3007";
+      proxyWebsockets = true;
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:29]: Is that necessary? I don't remember setting it up on my config.

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    environmentFile = "/path/to/my/secrets/file";
+  };
+
+  # This sets up a woodpecker agent
+  services.woodpecker-agents.agents."docker" = {
+    enable = true;
+    # We need this to talk to the podman socket
+    extraGroups = [ "podman" ];
+    environment = {
+      WOODPECKER_SERVER = "${domain}:9000";
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:54]: ... If you use localhost then no need to open ports.

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+  # Here we setup podman and enable dns
+  virtualisation.podman = {
+    enable = true;
+    defaultNetwork.settings = {
+      dns_enabled = true;
+    };
+  };
+  # This is needed for podman to be able to talk over dns
+  networking.firewall.interfaces."podman0" = {
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:71]: Is it `podman0` or `podman+` (as I remember reading in somebody else's config)?

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    allowedTCPPorts = [ 53 ];
+  };
+}
+```
+
+You can find all the configuration options [here](https://search.nixos.org/options?channel=unstable&size=200&sort=relevance&query=woodpecker)
+
+## Tips and tricks
+
+There is some resources on how to utilize woodpecker more effectively with nix in the [awesome](#awesome) section, like using the runners nix-store in the pipeline
```
Comment (ambroisie) [inline docs/docs/30-administration/22-backends/50-nixos.md:82]:
```suggestion
There are some resources on how to utilize woodpecker more effectively with nix in the [awesome](#awesome) section, like using the runners nix-store in the pipeline
```

**Review by ambroisie — COMMENTED** [review]
Some small feedback.

File: docs/docs/30-administration/22-backends/50-nixos.md (hunk 7 above)
Comment (Janik-Haag) [reply docs/docs/30-administration/22-backends/50-nixos.md:71]: By default it's podman0, the `podman+` or `podman*`, which one you have to use depends on your firewall backend (nftables vs iptables)

**Review by ambroisie — APPROVED** [review]

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    allowedTCPPorts = [ 53 ];
+  };
+}
+```
+
+You can find all the configuration options [here](https://search.nixos.org/options?channel=unstable&size=200&sort=relevance&query=woodpecker)
+
+## Tips and tricks
+
+There are some resources on how to utilize woodpecker more effectively with nix in the [awesome](#awesome) section, like using the runners nix-store in the pipeline
```
Comment (xoxys) [inline docs/docs/30-administration/22-backends/50-nixos.md:87]:
```suggestion
There are some resources on how to utilize Woodpecker more effectively with NixOS in the [awesome](#awesome) section, like using the runners nix-store in the pipeline
```

Comment (xoxys) [inline docs/docs/30-administration/22-backends/50-nixos.md:87]: The ref to the `aweseome` page looks wrong to me, do you mean `[awesome](../92-awesome.md)? In that case, I would also name it `Awesome Woodpecker` and call it a page not section.

**Review by xoxys — CHANGES_REQUESTED** [review]

File: docs/docs/30-administration/22-backends/50-nixos.md (hunk 9 above)
Comment (Janik-Haag) [reply docs/docs/30-administration/22-backends/50-nixos.md]: Yes you are right, I changed it.

**Comment by Janik-Haag** [comment]
>  Maybe we could add a new section "third-party installation methods" to make clear that this is not officially and bug reports related to the installation method should not be opened here.

Feel free to move it to some other section, if you feel like there is a better place. I also added a link to where to open issues so that hopefully the woodpecker team doesn't get bothered: 

:::info
Note that this module is not maintained by the woodpecker-developers.
If you experience issues please open a bug report in the [nixpkgs repo](https://github.com/NixOS/nixpkgs/issues/new/choose) where the module is maintained.
:::

**Comment by xoxys** [comment]
@qwerty287 do you want to have something changed before merge?

**Review by xoxys — APPROVED** [review]

**Comment by Janik-Haag** [comment]
The link to the awsome page is broken :/

File: docs/docs/30-administration/22-backends/50-nixos.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
+# NixOS Deployment
```
Comment (anbraten) [inline docs/docs/30-administration/22-backends/50-nixos.md:1]: NixOS is not a backend?!

Comment (Janik-Haag) [reply docs/docs/30-administration/22-backends/50-nixos.md:1]: Not really you deploy the go binary and utilize a backend like docker, local or anything else.

Comment (Janik-Haag) [reply docs/docs/30-administration/22-backends/50-nixos.md:1]: In the added example we use the docker backend

Comment (Janik-Haag) [reply docs/docs/30-administration/22-backends/50-nixos.md:1]: did I mess something up or do you have improvements?

**Pull request merged**